geometry/sat_collision.h

```diff
@@ -0,0 +1,251 @@
+/**
+* @file
+* @brief Implementation of the [Hyperplane Separation Theorem]
+* (https://en.wikipedia.org/wiki/Hyperplane_separation_theorem) (or
+* Separating Axis Theorem) for detecting collisions in 2D-space.
+* @details
+* Implements [Hyperplane Separation Theorem]
+* (https://en.wikipedia.org/wiki/Hyperplane_separation_theorem) (or
+* Separating Axis Theorem) to detect convex polygon-polygon and convex
+* polygon-circle collision.
+*
+* Two convex objects do not overlap if there exists a line (called axis) onto
+* which the two objects' projections do not overlap. The axis is a separating
+* axis if an orthogonal straight line (called the separating line) can be drawn
+* between the objects without colliding with either.
+*
+* The algorithm should run in O(n) time for polygon-X collisions and O(1) for
+* circle-circle collisions, where n is the largest number of vertices that
+* belongs to one of the polygons
+
+* @author [William Fors](https://github.com/WeRiano)
+*/
+
+namespace geometry {
+
+namespace collision {
```

Comment:
```suggestion
/**
 * @namespace geometry
 * @brief Geometry-related algorithms
 */
namespace geometry {
/**
 * @namespace collision
 * @brief Functions for the [Hyperplane Separation Theorem]
* (https://en.wikipedia.org/wiki/Hyperplane_separation_theorem) algorithm implementation
*/
namespace collision {
```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix